synfig-studio/src/gui/states/state_bone.cpp

```diff
@@ -729,148 +742,19 @@ StateBone_Context::event_mouse_release_handler(const Smach::event& x)
 								bone.set_angle((releaseOrigin - clickOrigin).angle());
 							}
 
-							ValueNode_Bone::Handle bone_node = ValueNode_Bone::create(bone,get_canvas());
-							action->set_param("item",ValueNode::Handle::cast_dynamic(bone_node));
-
-							action->set_param("value_desc",ValueDesc(list_node,0));
-
-							if(action->is_ready()){
-								try{
-									get_canvas_interface()->get_instance()->perform_action(action);
-								} catch (...) {
-									info("Error performing action");
-								}
-							}
-
-							Action::Handle setActiveBone(Action::Handle(Action::create("ValueNodeSetActiveBone")));
-							setActiveBone->set_param("canvas",get_canvas());
-							setActiveBone->set_param("canvas_interface",get_canvas_interface());
-
-							setActiveBone->set_param("active_bone_node",ValueNode::Handle::cast_dynamic(bone_node));
-							setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
-
-							if (setActiveBone->is_ready()){
-								try{
-									get_canvas_interface()->get_instance()->perform_action(setActiveBone);
-								} catch (...) {
-									info("Error performing action");
-								}
-							}
-						}
-
-					}
-				}
-				else if(deform_layer){ //!if selected layer is a Skeleton deform Layer and user wants to work on a skeleton deform layer
-					update_tool_options(1);
-					bool is_currently_on(get_work_area()->get_type_mask()&Duck::TYPE_WIDTH);
-					if(!is_currently_on){
-						get_canvas_view()->toggle_duck_mask(Duck::TYPE_WIDTH);
-					}
+							ValueNode::Handle new_active_bone_node;
+							if (skel_layer) {
+								new_active_bone_node = ValueNode_Bone::create(bone,get_canvas());
+								action->set_param("item",new_active_bone_node);
+							} else {
+								Bone bone2 = bone;
```

Comment:
It looks like you not need `bone2` anymore. 
So you can just use:
```
ValueNode_Composite::create(std::pair<Bone,Bone>(bone, bone)
```

Reply:
Done